### PR TITLE
Expose ProcedureGuard to check tx termination and guards from within …

### PR DESCRIPTION
**Pull request by boggle**
…long running procedures

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/guard/TimeoutGuard.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void check( KernelStatement statement )
         check( statement.getTransaction() );
     }
 
-    private void check( KernelTransactionImplementation transaction )
+    public void check( KernelTransactionImplementation transaction )
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/guard/TimeoutGuard.java]: Please add `@Override` here

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java [hunk 2]
```diff
@@ -421,6 +428,37 @@ private Procedures setupProcedures( PlatformModule platform, EditionModule editi
         return procedures;
     }
 
+    private ComponentRegistry.Provider<TerminationGuard> procedureGuard( final Dependencies dependencies )
+    {
+        Guard guard = dependencies.resolveDependency( Guard.class );
+        return (ctx) ->
+        {
+            KernelTransaction ktx = ctx.get( KERNEL_TRANSACTION );
+            return (TerminationGuard) () ->
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java]: please extract this class into separate class and put it into some procedure related package, most probably where `TerminationGuard` live

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
@boggle  Two small comments and also please note there is test failure atm. Review round complete

**Comment by boggle** [comment]
This was green; just rebased but since it's weekend waiting for another build.

**Pull request merged**